pranav/cart/spec.rb

```diff
@@ -0,0 +1,134 @@
+load "main.rb"
+
+descibe "cal_gst"
+  context "calculate GST on gross_amount" do
+    
+    before do
+      s_gst = 8
+      c_gst = 8
+    end
+    
+    it "return GST amount on gross_amount" 
+      true_amount = (gross_amount / 100) * (c_gst + s_gst)
+      gst_amount = cal_gst(amount)
+      expect(gst_amount).to eq(true_amount)
+    end
+
+  end  
+end   
+
+descibe "cal_discount"
+  context "calculate discount amount on gross_amount" do
+    it "return discount amount on gross_amount" 
+      true_amount = amount > 300 ? amount * 0.3 : amount > 200 ? amount * 0.2 : amount > 100 ? amount * 0.1 : amount * 0.05
+      discount_amount = cal_gst(amount)
+      expect(discount_amount).to eq(true_amount)
+    end
+  end  
+end  
+
+descibe "cal_gross_bill"
+  context "calculate gross_amount" do
+    before do
+      cart = { 1 => { name: 'bb', price: 30, quantity: 10 }, 2 => { name: 'cc', price: 40, quantity: 7 }} 
+    end 
+    
+    it "return  total gross_amount"       
+      expect(cal_gross_bill(cart)).to eq(580)
+    end
+  end  
+end  
+
+descibe "make_bill"
+  context "calculate gross_amount" do
+    before do
+      cart = { 1 => { name: 'bb', price: 30, quantity: 10 }, 2 => { name: 'cc', price: 40, quantity: 7 }} 
+    end 
+    
+    it "return  total gross_amount"       
+      expect(make_bill(cart)).to eq(498.79999999999995)
+    end
+  end  
+end  
+
+descibe "add_to_menu"
+  context "adding item to menu" do
+    before do
+      menu = { 1 => { name: 'aa', price: 10, quantity: 10 }, 2 => { name: 'bb', price: 30, quantity: 10 }, 3 => { name: 'cc', price: 40, quantity: 7 }} 
+    end 
+    
+    it "item gets added successfully, in menu key for that item id should return true"       
+      expect_any_instance_of(Kernel).to receive(:gets).and_return("apple")
+      expect_any_instance_of(Kernel).to receive(:gets).and_return(60)
+      expect_any_instance_of(Kernel).to receive(:gets).and_return(15)
+      # add_to_menu("apple", menu, 60, 15)
```

Comment:
you can use single statement for mocking gets like here it can be expect_any_instance_of(Kernel).to receive(:gets).and_return("apple\n", "60\n", "15\n")